src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -159,180 +101,9 @@
-   pool :- jruby-schemas/pool-queue-type]
-  (let [instance (borrow-fn pool)]
-    (cond (instance? PoisonPill instance)
-          (do
-            (.releaseItem pool instance)
-            (throw (IllegalStateException.
-                     "Unable to borrow JRuby instance from pool"
-                     (:err instance))))
-
-          (jruby-schemas/jruby-puppet-instance? instance)
-          instance
-
-          ((some-fn nil? jruby-schemas/retry-poison-pill?) instance)
-          instance
-
-          (instance? ShutdownPoisonPill instance)
-          instance
-
-          :else
-          (throw (IllegalStateException.
-                   (str "Borrowed unrecognized object from pool!: " instance))))))
-
-(schema/defn ^:always-validate
-  borrow-from-pool :- jruby-schemas/JRubyPuppetInstanceOrPill
-  "Borrows a JRubyPuppet interpreter from the pool. If there are no instances
-  left in the pool then this function will block until there is one available."
-  [pool-context :- jruby-schemas/PoolContext]
-  (borrow-from-pool!* borrow-without-timeout-fn
-                      (get-pool pool-context)))
-
-(schema/defn ^:always-validate
-  borrow-from-pool-with-timeout :- jruby-schemas/JRubyPuppetBorrowResult
-  "Borrows a JRubyPuppet interpreter from the pool, like borrow-from-pool but a
-  blocking timeout is provided. If an instance is available then it will be
-  immediately returned to the caller, if not then this function will block
-  waiting for an instance to be free for the number of milliseconds given in
-  timeout. If the timeout runs out then nil will be returned, indicating that
-  there were no instances available."
-  [pool-context :- jruby-schemas/PoolContext
-   timeout :- schema/Int]
-  {:pre  [(>= timeout 0)]}
-  (borrow-from-pool!* (partial borrow-with-timeout-fn timeout)
-                      (get-pool pool-context)))
-
-(schema/defn ^:always-validate
-  return-to-pool
-  "Return a borrowed pool instance to its free pool."
-  [instance :- jruby-schemas/JRubyPuppetInstanceOrPill]
-  (if (jruby-schemas/jruby-puppet-instance? instance)
-    (let [new-state (swap! (:state instance)
-                           update-in [:borrow-count] inc)
-          {:keys [max-requests flush-instance-fn pool]} instance]
-      (if (and (pos? max-requests)
-               (>= (:borrow-count new-state) max-requests))
-        (do
-          (log/infof (str "Flushing JRuby instance %s because it has exceeded the "
-                          "maximum number of requests (%s)")
-                     (:id instance)
-                     max-requests)
-          (try
-            (flush-instance-fn pool instance)
-            (finally
-              (.releaseItem pool instance false))))
-        (.releaseItem pool instance)))
-    ;; if we get here, it was from a borrow and we got a Retry, so we just
-    ;; return it to the pool.
-    (.releaseItem (:pool instance) instance)))
-
 (schema/defn ^:always-validate new-main :- jruby-schemas/JRubyMain
```

Comment:
Probably some other stuff that could go away along with this now, too, like `init-jruby-config` and `SupportedJRubyCompileModes`.